src/components/Pages/Admin/Designer.vue

```diff
@@ -573,15 +584,94 @@ export default {
                 },
             }]);
             this.$log.debug('groupId', groupId);
+
             await this.addElements(groupId);
         },
+
+        produceXMLWithoutConstructionInside(xml) {
+            const parser = new DOMParser();
+
+            const xmlDoc = parser.parseFromString(xml, 'text/xml');
+            const construction = xmlDoc.getElementsByTagName('construction')[0];
+
+            xmlDoc.documentElement.removeChild(construction);
+            this.$log.debug(xmlDoc);
+
+            const xmlText = new XMLSerializer().serializeToString(xmlDoc);
+            return xmlText;
+        },
+
+        extractPerspectives(xml) {
+            const parser = new DOMParser();
+            const xmlDoc = parser.parseFromString(xml, 'text/xml');
+
+            const visibleViews = Array.from(xmlDoc.getElementsByTagName('view'))
+                .filter(el => el.getAttribute('visible') === 'true');
+
+            // We sort the views so that we can later send the ordered
+            // arrangement of the different view tabs present
+            // in the activity designer to the students' views
+            const visibleViewsSorted = visibleViews.sort((a, b) => {
+                const aOrder = a.getAttribute('location').replace(/,/g, '');
+                const bOrder = b.getAttribute('location').replace(/,/g, '');
+
+                if (aOrder > bOrder) {
+                    return -1;
+                /* eslint-disable-next-line no-else-return */
+                } else if (bOrder > aOrder) {
+                    return 1;
+                } else {
+                    return 0;
+                }
+            });
+            // The following loop creates a string of the (encoded)
+            // values of the different views present in the
+            // activity designer (to be sent to the students)
+            let perspectivesMapped = '';
+
+            for (let i = 0; i < visibleViewsSorted.length; i += 1) {
```

Comment:
Explain in the docstring where did you get this mapping between integers and letters.